server/src/main/java/org/elasticsearch/gateway/Gateway.java

```diff
@@ -41,22 +44,27 @@
 
     private final ClusterService clusterService;
 
-    private final TransportNodesListGatewayMetaState listGatewayMetaState;
+    private final NodeClient client;
 
     private final IndicesService indicesService;
 
     public Gateway(final Settings settings, final ClusterService clusterService,
-                   final TransportNodesListGatewayMetaState listGatewayMetaState,
+                   final NodeClient client,
                    final IndicesService indicesService) {
         this.indicesService = indicesService;
         this.clusterService = clusterService;
-        this.listGatewayMetaState = listGatewayMetaState;
+        this.client = client;
     }
 
     public void performStateRecovery(final GatewayStateRecoveredListener listener) throws GatewayException {
-        DiscoveryNode[] discoveryNodes = clusterService.state().nodes().getMasterNodes().values().toArray(DiscoveryNode.class);
-        LOGGER.trace("performing state recovery from {}", discoveryNodes);
-        final TransportNodesListGatewayMetaState.NodesGatewayMetaState nodesState = listGatewayMetaState.list(discoveryNodes, null).actionGet();
+        final DiscoveryNode[] nodes = clusterService.state().nodes().getMasterNodes().values().toArray(DiscoveryNode.class);
```

Comment:
We changed this in  https://github.com/crate/crate/commit/ed18312ee5c6331a3dce68bd5b628a71388ac484

Maybe we could have gone the other way to avoid streaming the bigger DiscoveryNode. Maybe something to revisit later.

Reply:
I see, thanks for the info, wasn't aware of this.

Reply:
I also forgot about it until I had to look why the cherry-pick conflicted :)